pkg/repository/mirror.go

```diff
@@ -219,6 +219,7 @@ L:
 		return nil, errors.Annotatef(err, "download from %s failed", url)
 	}
 	if maxSize > 0 && resp.BytesComplete() > maxSize {
+		err = errors.New("maximum size exceeded")
```

Comment:
By design?

Reply:
Yes, otherwise the result of `errors.Annotatef(err, "download from %s failed, maximum size exceeded", url)` will be nil, and the download method will returns nil, nil.